library/src/scala/caps/package.scala

```diff
@@ -0,0 +1,155 @@
+package scala
+package caps
+
+import annotation.{experimental, compileTimeOnly, retainsCap}
+
+/**
+ * Base trait for classes that represent capabilities in the
+ * [object-capability model](https://en.wikipedia.org/wiki/Object-capability_model).
+ *
+ * A capability is a value representing a permission, access right, resource or effect.
+ * Capabilities are typically passed to code as parameters; they should not be global objects.
+ * Often, they come with access restrictions such as scoped lifetimes or limited sharing.
+ *
+ * An example is the [[scala.util.boundary.Label Label]] class in [[scala.util.boundary]].
+ * It represents a capability in the sense that it gives permission to [[scala.util.boundary.break break]]
+ * to the enclosing boundary represented by the `Label`. It has a scoped lifetime, since breaking to
+ * a `Label` after the associated `boundary` was exited gives a runtime exception.
+ *
+ * [[Capability]] has a formal meaning when
+ * [[scala.language.experimental.captureChecking Capture Checking]]
+ * is turned on.
+ * But even without capture checking, extending this trait can be useful for documenting the intended purpose
+ * of a class.
+ */
+trait Capability extends Any
```

Comment:
Please keep this `@experimental` in this PR, which should be purely a refactoring. Then, once the refactoring is merged, make a separate PR that actually makes this non experimental. Don't hide the significant needle in a huge haystack of refactoring.